### PR TITLE
Add valkyrie base

### PR DESCRIPTION
**Pull request by sbranden**
Patch series to add base Valkyrie SoC support in first commit and base support for bcm958401m2 board based on Valkyrie SoC in second commit.  Board boots and runs hello_world app with console support. 
 Future commits will add additional functionality to Valkyrie SoC as well as board functionality.

### CODE REVIEW

**Comment by sbranden** [comment]
Please provide guidelines for any missing documentation that is required.  I could not find guidelines anywhere for SoC nor Board addition requirements in order to upstream.  But I do see yaml files elsewhere but don't know which ones are actually required.

**Comment by sbranden** [comment]
Could somebody please explain the build failure to me.  Looking at the build console output I do not know how to build such test locally to debug the issue?

**Review by galak — DISMISSED** [review]
board port seems to be missing any docs.

**Comment by sbranden** [comment]
> board port seems to be missing any docs.

missing boards/arm/bcm958401m2/doc/index.rst added and doc/conf.py modified.

**Comment by sbranden** [comment]
Do I need to keep rebasing this to latest or will you do with that when merged?

**Comment by sbranden** [comment]
@MaureenHelm @ioannisg - would be great for you to review?

File: boards/arm/bcm958401m2/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+Board is booted over PCIe interface.
+
+Debugging
+=========
+The bcm958401m2 board includes pads for soldering a JTAG connector.
+Zephyr applications running on the M7 core can also be tested by observing UART console output.
+
+
+References
+**********
```
Comment (galak) [inline boards/arm/bcm958401m2/doc/index.rst]: Any ability to put a reference to the board?  Ie some URL for the board?

File: dts/arm/broadcom/valkyrie.dtsi [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+			mpu: mpu@e000ed90 {
+				compatible = "arm,armv7m-mpu";
+				reg = <0xe000ed90 0x40>;
+				arm,num-mpu-regions = <16>;
+			};
+		};
+	};
+
+	sram0: memory@400000 {
```
Comment (galak) [inline dts/arm/broadcom/valkyrie.dtsi]: I assume the sram is part of the SoC?  If so can you move this node inside the SoC container?

**Review by galak — COMMENTED** [review]
LGTM.  Some minor comments.

File: doc/conf.py [hunk 3]
```diff
@@ -315,6 +315,7 @@
 		('boards/arm/arduino_due/doc/board', 'boards/arm/arduino_due/doc/index'),
 		('boards/arm/arduino_zero/doc/arduino_zero', 'boards/arm/arduino_zero/doc/index'),
 		('boards/arm/atsamd20_xpro/doc/atsamd20_xpro', 'boards/arm/atsamd20_xpro/doc/index'),
+		('boards/arm/bcm958401m2/doc/board', 'boards/arm/bcm958401m2/doc/index'),
```
Comment (MaureenHelm) [inline doc/conf.py]: I don't think this is necessary

**Review by MaureenHelm — DISMISSED** [review]
one nit, otherwise looks good

**Comment by galak** [comment]
> Do I need to keep rebasing this to latest or will you do with that when merged?

only if you make a change would I bother to rebase, or if there is some reported merge conflict.

File: doc/conf.py (hunk 3 above)
Comment (sbranden) [reply doc/conf.py]: > I don't think this is necessary

I can remove if you want.  But why are all the other boards listed here?

I would think this is needed for having the board added to the board documentation?  Perhaps it is used to generate an html page like this?
https://docs.zephyrproject.org/latest/boards/index.html

File: dts/arm/broadcom/valkyrie.dtsi (hunk 2 above)
Comment (sbranden) [reply dts/arm/broadcom/valkyrie.dtsi]: > I assume the sram is part of the SoC? If so can you move this node inside the SoC container?

Yes, sram is internal to the SoC.  I will move this node instead the SoC container.

File: boards/arm/bcm958401m2/doc/index.rst (hunk 1 above)
Comment (sbranden) [reply boards/arm/bcm958401m2/doc/index.rst]: There is no URL referencing this board at this time.  It is not a consumer product with such marketing and is used and sold directly without information posted on the internet.

File: doc/conf.py (hunk 3 above)
Comment (galak) [reply doc/conf.py]: I thinks it was for some transition in how we structured docs a while ago.  It shouldn't be needed, please remove.

Comment (sbranden) [reply doc/conf.py]: > I thinks it was for some transition in how we structured docs a while ago. It shouldn't be needed, please remove.

Ok thanks - change to doc/conf.py removed.  Sounds like the entire doc/conf.py file should be removed if it is not needed now then??

**Review by sbranden — COMMENTED** [review]
all comments from review addressed - thanks.

**Review by galak — APPROVED** [review]

**Comment by sbranden** [comment]
Build failed due to shippable account failure.  Doesn't seem related to my change?  But I rebase and pushed to trigger another build verification.

**Pull request merged**